CONTRIBUTING.md

```diff
@@ -0,0 +1,67 @@
+
+If you're adding new features to **editorconfig-checker**, please include tests.
+
+### Review process and requirements to merge
+
+To ensure everyone in the community has the opportunity to share their feedback, ideas, and suggestions, a Pull Request must go through a review process, and must meet the following requirements to be merged:
+
+- **Approval:** At least **2 core team members** must approve the Pull Request.
+- **Open Duration:** The Pull Request must be **open for at least 72 hours**.
+- **CI checks:** The Pull Request must **pass all CI checks**, including tests and build.
+
+**Note:** If a critical bug fix is required, and the Pull Request was approved by at least **3 core team members**, the Pull Request **can be merged before the 72 hours** open duration.
+
+Pull requests are merged using the `Squash and Merge` option to keep the commit history clean, and promote clear separation of changes in separated pull requests.
+
+This collaborative process ensures that all perspectives are considered, and helps improve the project quality as a whole.
+
+## Commits and Releases
+
+The commit message guidelines adheres to [Conventional Commits](https://www.conventionalcommits.org/) and [Semantic Versioning](https://semver.org/) for releases.
+
+### Semantic Versioning Policy
+
+As **editorconfig-checker** is a code quality tool, it's not always clear when a minor or major version bump occurs. The following rules are used to determine the version bump:
+
+- Patch release (1.0.x -> 1.0.y)
+  - Updates to format outputs (error messages, logs, ...).
+  - Performance improvements which doesn't affect behavior.
+  - Build process changes (e.g. updating dependencies, updating `Dockerfile`, ...).
+  - Reverts (reverting a previous commit).
+  - Bug fixes which result in **editorconfig-checker** reporting less linting errors, also known as "false positives".
+- Minor release (1.x.0 -> 1.y.0)
+  - Adding new [configuration](./README.md#configuration) options, including new CLI flags.
+  - Adding new [default paths](./README.md#default-excludes) to exclude by default.
+  - Adding new [format outputs](./README.md#output-formats).
+  - Support a new [editorconfig](https://editorconfig.org/) property (e.g: `insert_final_newline`, `indent_size`, ...).
+  - Any new feature which doesn't break existing behavior.
+- Major release (x.0.0 -> y.0.0)
+  - Removal/Rename of a [configuration](./README.md#configuration) option.
+  - Removal of a [format outputs](./README.md#output-formats).
+  - Removal of a [default paths](./README.md#default-excludes) to exclude by default.
+  - Removal of a [editorconfig](https://editorconfig.org/) property.
+  - Bug fixes, which might result in **editorconfig-checker** reporting more linting errors, if the previous behavior was incorrect, with the [editorconfig](https://editorconfig.org/) specification.
```

Comment:
It has already been discussed previously: https://github.com/editorconfig-checker/editorconfig-checker/pull/350#issuecomment-2203634216

We might consider anything that results in **editorconfig-checker** reporting more linting errors, a major version.